### PR TITLE
Release config

### PR DESCRIPTION
**Pull request by floryst**
This MR adds the semantic-release configuration to itk-wasm. The configuration details are provided below.
- Merges and commits to the `release` branch will publish to the default distribution channel, i.e. `npm install itk-wasm`. The tag format is `itk-wasm-v{version}`.
  - After this MR, adding a commit to the `release` branch will release itk-wasm v1.0.0.
- Merges and commits to the `master` branch will publish to the beta distribution channel, i.e. `npm install itk-wasm@beta`. The tag format is `itk-wasm-v{version}.b{betaVersion}`.
  - After this MR, adding a major/minor/patch commit to the `master` branch will release itk-wasm v1.0.0.b9, since the current beta version is b8.
- Semantic release will only run on the `master`, `release`, and any designated maintenance branches.
- There is no concept of an alpha release on master. If alpha releases are desired in the future, then a new branch will be needed.

### CODE REVIEW

File: .github/workflows/javascript-typescript.yml [hunk 1]
```diff
@@ -66,6 +66,15 @@ jobs:
 
         cd -
 
+    - name: Publish task
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (floryst) [inline .github/workflows/javascript-typescript.yml:72]: @thewtex you will need to add an NPM_TOKEN to the repo secrets in order to publish.

Do you want to also publish to github? If not, we can disable the github plugin.

File: .github/workflows/javascript-typescript.yml [hunk 2]
```diff
@@ -66,6 +66,15 @@ jobs:
...
+      run: |
+        git config --global user.name "Github Actions"
+        git config --global user.email "matt.mccormick@kitware.com"
```
Comment (floryst) [inline .github/workflows/javascript-typescript.yml:75]: Is there an itk robot account?

**Comment by floryst** [comment]
Another note: due to how the beta channel works, adding more commits to `master` will result in `npm install itk-wasm` not installing the latest updates. That will have to be through `npm install itk-wasm@beta`.

File: .github/workflows/javascript-typescript.yml (hunk 1 above)
Comment (thewtex) [reply .github/workflows/javascript-typescript.yml:72]: :1st_place_medal: :rocket: :man_singer: 

NPM_TOKEN added.

Sure, let's publish to GitHub, too.

File: .github/workflows/javascript-typescript.yml (hunk 2 above)
Comment (thewtex) [reply .github/workflows/javascript-typescript.yml:75]: `Github` -> `GitHub`

We can use `itk+community@discourse.itk.org` for the email address.

**Review by thewtex — CHANGES_REQUESTED** [review]
@floryst looks great, thanks!!

We are now done with alpha's, the new beta naming scheme sounds good to me.

I minor request inline.

**Review by thewtex — APPROVED** [review]
Thank you, @floryst !!

**Pull request merged**